main/lsp/requests/definition.cc

```diff
@@ -35,16 +35,21 @@ unique_ptr<ResponseMessage> DefinitionTask::runRequest(LSPTypecheckerDelegate &t
             for (auto loc : sym.locs(gs)) {
                 locMapping.emplace_back(loc, config.loc2Location(gs, loc));
             }
-            // Move all non-existent Locations to the front of the vector.
-            auto validLocations = absl::c_partition(locMapping, [](const auto &p) { return p.second == nullptr; });
-            // If we have multiple locations, eliminate "definitions" in RBI files
-            // for classes and modules, since the one(s) in Ruby files are more
-            // likely to be what the user is looking for.
-            if (sym.isClassOrModule() && std::distance(validLocations, locMapping.end()) > 1) {
-                validLocations = std::partition(validLocations, locMapping.end(),
-                                                [&gs](const auto &p) { return p.first.file().data(gs).isRBI(); });
+            {
+                // Erase all invalid locations
+                auto validLocations = absl::c_partition(locMapping, [](const auto &p) { return p.second != nullptr; });
```

Comment:
OOC, would it be the same if you use `remove_if` instead of `partion`?

Reply:
yes, but there is no `absl::c_remove_if` it looks like so I just used this because I didn't want to have to look up `std::remove_if` docs.